test/lightning_web/live/project_live_test.exs

```diff
@@ -4956,6 +4956,397 @@ defmodule LightningWeb.ProjectLiveTest do
+
+      # fill in the form and submit with the same name
+      view
+      |> form("#create-collection-modal form",
+        collection: %{raw_name: collection.name}
+      )
+      |> render_change()
+
+      html =
+        view
+        |> form("#create-collection-modal form")
+        |> render_submit()
+
+      assert html =~ "A collection with this name already exists"
+    end
+
+    test "shows an error when the limiter returns an error", %{
+      conn: conn,
+      user: user
+    } do
+      project = insert(:project)
+
+      conn = setup_project_user(conn, project, user, :owner)
+
+      error_msg = "Some error message"
+
+      Mox.stub(Lightning.Extensions.MockCollectionHook, :handle_create, fn
+        _attrs ->
+          {:error, :exceeds_limit, %{text: error_msg}}
+      end)
+
+      {:ok, view, _html} =
+        live(
+          conn,
+          ~p"/projects/#{project.id}/settings#collections"
+        )
+
+      # open the modal
+      view |> element("#open-create-collection-modal-button") |> render_click()
+
+      # modal is now present
+      assert has_element?(view, "#create-collection-modal")
+
+      # fill in the form and submit with the same name
+      view
+      |> form("#create-collection-modal form",
+        collection: %{raw_name: "some name"}
+      )
+      |> render_change()
+
+      view
+      |> form("#create-collection-modal form")
+      |> render_submit()
+
+      flash =
+        assert_redirected(
+          view,
+          ~p"/projects/#{project.id}/settings#collections"
+        )
+
+      assert flash["error"] == error_msg
+    end
+
+    test "user can edit a collection successfully", %{
+      conn: conn
+    } do
+      project = insert(:project)
+
+      for {conn, _user} <- setup_project_users(conn, project, [:viewer]) do
+        collection = insert(:collection, project: project)
+
+        {:ok, view, _html} =
+          live(
+            conn,
+            ~p"/projects/#{project.id}/settings#collections"
+          )
+
+        button = element(view, "#edit-collection-#{collection.id}-button")
+        assert has_element?(button)
+
+        # modal is not present
+        refute has_element?(view, "#edit-collection-#{collection.id}-modal")
+
+        # try clicking the button
+        assert_raise ArgumentError, ~r/is disabled/, fn ->
+          render_click(button)
+        end
+
+        # send event either way
+        view
+        |> with_target("#collections")
+        |> render_click("toggle_action", %{
+          "action" => "edit",
+          "collection" => collection.name
+        })
+
+        flash =
+          assert_redirected(
+            view,
+            ~p"/projects/#{project.id}/settings#collections"
+          )
+
+        assert flash["error"] == "You are not authorized to perform this action"
+      end
+
+      for {conn, _user} <-
+            setup_project_users(conn, project, [:owner, :admin, :editor]) do
```

Comment:
This is a nice coverage but would you mind moving to a separate test case for the different groups of users? One for the viewers and another one for different expected result contemplating this second group of users.

Reply:
Yeah, let me move them out. I did it that way for the `create` test and got lazy when handling `edit` and `delete` 😅